spyder/plugins/projects.py

```diff
@@ -153,6 +154,9 @@ def register_plugin(self):
         self.sig_project_closed.connect(
             lambda v: self.editor.setup_open_files())
         self.recent_project_menu.aboutToShow.connect(self.setup_menu_actions)
+        self.run.connect(lambda fname: self.main.open_external_console(

```

Comment:
@andfoy, let's make the file to be run in a dedicated console (instead of in an external one). I think it makes more sense (at least to me).

The problem is external consoles are immediately closed, unless you change the run options for your file.